components/script/dom/html/htmlmediaelement.rs

```diff
@@ -1637,6 +1637,54 @@ impl HTMLMediaElement {
         self.select_next_source_child();
     }
 
+    /// <https://html.spec.whatwg.org/multipage/#media-data-processing-steps-list>
+    /// => "If the media data cannot be fetched at all, due to network errors..."
+    /// => "If the media data can be fetched but is found by inspection to be in an unsupported
+    /// format, or can otherwise not be rendered at all"
+    fn media_data_processing_failure_steps(&self) {
+        // Step 1. The user agent should cancel the fetching process.
+        if let Some(ref mut current_fetch_context) = *self.current_fetch_context.borrow_mut() {
+            current_fetch_context.cancel(CancelReason::Error);
+        }
+
+        // Step 2. Abort this subalgorithm, returning to the resource selection algorithm.
+        self.resource_selection_algorithm_failure_steps();
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#media-data-processing-steps-list>
+    /// => "If the connection is interrupted after some media data has been received..."
+    /// => "If the media data is corrupted"
+    fn media_data_processing_fatal_steps(&self, error: u16) {
+        *self.source_children_pointer.borrow_mut() = None;
+        self.current_source_child.set(None);
+
+        // Step 1. The user agent should cancel the fetching process.
+        if let Some(ref mut current_fetch_context) = *self.current_fetch_context.borrow_mut() {
+            current_fetch_context.cancel(CancelReason::Error);
+        }
+
+        // Step 2. Set the error attribute to the result of creating a MediaError with
+        // MEDIA_ERR_NETWORK/MEDIA_ERR_DECODE.
+        self.error.set(Some(&*MediaError::new(
+            &self.owner_window(),
+            error,
+            CanGc::note(),
```

Comment:
Please add a `CanGc` argument to this method and use that instead of each use of CanGc::note() in this method.